graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/ScriptingApiResource.java

```diff
@@ -0,0 +1,119 @@
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static org.graylog2.shared.rest.documentation.generator.Generator.CLOUD_VISIBLE;
+
+@Api(value = "ScriptingApi", tags = {CLOUD_VISIBLE})
+@Path("/scripting_api")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes({MediaType.APPLICATION_JSON})
+@RequiresAuthentication
+public class ScriptingApiResource extends RestResource implements PluginRestResource {
+    private static final Logger LOG = LoggerFactory.getLogger(ScriptingApiResource.class);
+
+    private final SearchExecutor searchExecutor;
+    private final EventBus serverEventBus;
+    private final SearchRequestSpecToSearchMapper searchCreator;
+    private final SearchTypeResultToTabularResponseMapper responseCreator;
+
+    @Inject
+    public ScriptingApiResource(final SearchExecutor searchExecutor,
+                                final EventBus serverEventBus,
+                                final SearchRequestSpecToSearchMapper searchCreator,
+                                final SearchTypeResultToTabularResponseMapper responseCreator) {
+        this.searchExecutor = searchExecutor;
+        this.serverEventBus = serverEventBus;
+        this.searchCreator = searchCreator;
+        this.responseCreator = responseCreator;
+    }
+
+    @POST
+    @ApiOperation(value = "Execute aggregation specified by `searchRequestSpec`",
+                  response = TabularResponse.class)
+    @Path("aggregate")
+    @NoAuditEvent("Creating audit event manually in method body.")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response executeQuery(@ApiParam(name = "searchRequestSpec") @Valid SearchRequestSpec searchRequestSpec,
+                                 @Context SearchUser searchUser) {
+
+        //Step 1: map simple request to more complex search
+        Search search = searchCreator.apply(searchRequestSpec);
+
+        //Step 2: execute search as we usually do
+        final SearchJob searchJob = searchExecutor.execute(search, searchUser, ExecutionState.empty());
+        postAuditEvent(searchJob);
+
+        //Step 3: take complex response and try to map it to simpler, tabular form
+        final SearchJobDTO searchJobDTO = SearchJobDTO.fromSearchJob(searchJob);
+        final QueryResult queryResult = searchJobDTO.results().get(SearchRequestSpecToSearchMapper.QUERY_ID);
+        if (queryResult != null) {
+            final SearchType.Result aggregationResult = queryResult.searchTypes().get(AggregationSpecToPivotMapper.PIVOT_ID);
+            if (aggregationResult instanceof PivotResult pivotResult) {
+                final TabularResponse tabularResponse = responseCreator.mapToResponse(searchRequestSpec, pivotResult);
+                return Response.ok(tabularResponse).build();
```

Comment:
We can just return `tabularResponse` in here, that makes it a bit easier to understand from the method signature what's returned. Below, we can throw a `NotFoundException`.